### PR TITLE
Frontend web development react angular vue : React Testing Library

### LINKED ISSUE #8144: Frontend_with_Framework: React 9.4 -> React Testing Library
### Description

Contributors are required to create Documentation and explain about the **React Testing Library** which should cover the question like :

- What is React Testing Library ?
- Where we use ?
- What are the benefits of React Testing Library ?
- also give some theory explanation (with code samples) .
- also any add-ons in depth about these topic from your side.

Note :

- Plagiarism is strictly prohibited.
- **Changes should be made inside the Frontend_Web_Development_React_Angular_Vue/ dir inside Frontend_Web_Development_React_Angular_Vue branch.**
- Files need to be added into the **Week 9** folder only.

### Domain

Frontend Dev React/Angular/Vue

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by its-me-Harsh-Anand**
<hr>

## Description 📜
- What is React testing library
- Where to use it
- Benefits of react testing library
- Short-comings of React Testing library
- Methods for finding elements
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. 
Fixes #8144 

<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] Frontend Dev React/Angular/Vue

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Review by heet-kumar — APPROVED** [review]
@its-me-Harsh-Anand ok LGTM !! 👍 now there is not much time left and lets wait for the supervisor approval and if they might be asked you to make changes then make that to.

**Comment by heet-kumar** [comment]
@anandhu720 review this PR once from your side.

**Review by anandhu720 — APPROVED** [review]
LGTM

**Pull request merged**
